addons/sale/models/sale_order.py

```diff
@@ -597,6 +602,14 @@ def _onchange_commitment_date(self):
                 }
             }
 
+    @api.depends('company_id', 'partner_credit', 'partner_credit_limit', 'amount_total')
```

Comment:
```suggestion
    @api.depends('company_id', 'amount_total')
```

Reply:
Now that `amount_total` is a dependency, the compute is triggered correctly, even without `partner_credit` and `partner_credit_limit` :ok_hand:

Reply:
But I had to put them back once again :confounded: 
Otherwise, the banner was not triggered when switching partners...

Reply:
depends on 'partner_id' then ? :D

Reply:
OK :slightly_smiling_face: 
I thought the `@api.depends` was "only" for fields that were explicitly referenced in the compute :sweat_smile:

Reply:
For stored computes, you need all dependencies which should trigger the field recomputation, but for 'UX' computes, you should only use the strictly necessary one for your UX uses.

Suppose you modify the `credit_limit` of a partner, if there is a dependency, the ORM will search in database all the SO using this partner, even if invalidating & recomputing the field is useless because it's not stored.  The ORM has to do that because he doesn't know that this field is only used in a view and never in a business logic, so he has to know and check whether there are records for which the fields has to be recomputed.  Unless somebody reads the field, I think it won't be computed, but at least the search is "uselessly" triggered.

Reply:
OK I see :slightly_smiling_face: 
Thanks for the explanation :ok_hand: :

Reply:
> Suppose you modify the `credit_limit` of a partner, if there is a dependency, the ORM will search in database all the SO using this partner, even if invalidating & recomputing the field is useless because it's not stored. 

Doesn't the ORM treat stored/non stored fields separately?

> The ORM has to do that because he doesn't know that this field is only used in a view and never in a business logic, so he has to know and check whether there are records for which the fields has to be recomputed. Unless somebody reads the field, I think it won't be computed, but at least the search is "uselessly" triggered.

AFAIK, the ORM will only drop the value from the cache, without searching for anything, which isn't that expensive I think.
Then it will recompute it only if the field is requested again. (and it that case it should recompute it since the dependencies have changed...)

Reply:
I might have a confused version of the situation in my mind because of a recent perf problem (but in my case, the computed field was considered stored by the ORM because specified on a SQL view).
I'll verify this point :eyes:.

Reply:
> Doesn't the ORM treat stored/non stored fields separately?

Install base, modify the code of a random res.country record:
* the ORM will search all partners with the country (because of the `country_code` field on partners, which is not stored)
* if account is also installed, it will search companies with the specified country (because of the `country_code` field added in `account` on `res.company`, not stored either, how could it be if the base field is not anyway)
* and other cases according to installed modules, some also with a logger error because the `res.country` field is not stored
 
> AFAIK, the ORM will only drop the value from the cache, without searching for anything, which isn't that expensive I think. Then it will recompute it only if the field is requested again. (and it that case it should recompute it since the dependencies have changed...)

It searches for records, but for the cache things, I suppose it's correctly managed and even though the field is marked as 'to recompute' on X records (after the search), it won't be computed (except if read) because not stored (and therefore no need to propagate it to the database).